### PR TITLE
Add docker-tox image for building things

### PR DESCRIPTION
**Pull request by alexcouper**
A useful image that will end up in docker hub that means in jenkins users can easily create containers that inherit from this one and have tox and multiple python versions available.

Very debatable as to whether this should live elsewhere.

### CODE REVIEW

**Comment by alexcouper** [comment]
Once merged, this will release will fail and I'll solve the docker pushing on master.

#selfmerge

File: docker-tox/setup.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+PYTHON_VERSIONS=(2.7.10 3.5.2)
+PYTHON_GLOBAL_VERSION=2.7.10
+PYTHON_PIP_VERSION=7.1.2
```
Comment (txels) [inline docker-tox/setup.sh:5]: I'm curious: why not 8.1.2?

File: docker-tox/setup.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+# Install pyenv
+curl -L https://raw.githubusercontent.com/yyuu/pyenv-installer/master/bin/pyenv-installer -o /pyenv-installer
```
Comment (txels) [inline docker-tox/setup.sh:8]: Feels dangerous to install things out of the internet that can actually be a moving target.

**Review by txels — COMMENTED** [review]
Re: place to live, yes this could arguably live in its own repo like other docker images we have open sourced, and be deployed to dockerhub.

What was your main motivation for adding it here? Do you plan to support some built-in python build process based on tox directly in the builder?

**Comment by alexcouper** [comment]
> What was your main motivation for adding it here? Do you plan to support some built-in python build process based on tox directly in the builder?

Main motivation was that:
a) this repo claims to support a docker release workflow and so having something proove that seemed like a good idea
b) it is highly related to the component builder in that once we have this, we could support declaration of wanting a python version for a component and run things in docker
c) it was easy because it uses the compbuilder (i know that it will build on merge, and tag and push to docker etc)

**Pull request merged**